src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,143 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> firstNode;
+    private Node<T> lastNode;
+    private int size;
+
     @Override
     public void add(T value) {
+        if (firstNode == null) {
+            Node<T> newNode = new Node<>(null, value, null);
+            firstNode = newNode;
+            lastNode = newNode;
+        } else {
+            lastNode.next = new Node<>(lastNode, value, null);
+            lastNode = lastNode.next;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == size) {
+            add(value);
+            return;
+        } else if (index == 0) {
```

Comment:
```suggestion
        } 
        if (index == 0) {
```
we don't need else, we return from function on line 27